deployment/7_upgrade_v2.js

```diff
@@ -0,0 +1,133 @@
+const NftTransfer = require('../build/NftTransfer');
+const ModuleRegistry = require('../build/ModuleRegistry');
+const MultiSig = require('../build/MultiSigWallet');
+const Upgrader = require('../build/SimpleUpgrader');
+
+const utils = require('../utils/utilities.js');
+const DeployManager = require('../utils/deploy-manager.js');
+const MultisigExecutor = require('../utils/multisigexecutor.js');
+const semver = require('semver');
+
+const TARGET_VERSION = "1.1.0";
+const MODULES_TO_ENABLE = ["NftTransfer"];
+const MODULES_TO_DISABLE = [];
+const BACKWARD_COMPATIBILITY = 1;
+
+const deploy = async (network) => {
+
+    const newModuleWrappers = [];
+    const newVersion = {};
+
+    ////////////////////////////////////
+    // Setup
+    ////////////////////////////////////
+
+    const manager = new DeployManager(network);
+	await manager.setup();
+
+	const configurator = manager.configurator;
+	const deployer = manager.deployer;
+    const abiUploader = manager.abiUploader;
+    const versionUploader = manager.versionUploader;
+    const deploymentWallet = deployer.signer;
+    const config = configurator.config;
+
+    const ModuleRegistryWrapper = await deployer.wrapDeployedContract(ModuleRegistry, config.contracts.ModuleRegistry);
+    const MultiSigWrapper = await deployer.wrapDeployedContract(MultiSig, config.contracts.MultiSigWallet);
+    const multisigExecutor = new MultisigExecutor(MultiSigWrapper, deploymentWallet, config.multisig.autosign);
+    
+    console.log('Config:', config);
+
+    ////////////////////////////////////
+    // Deploy new modules
+    ////////////////////////////////////
+
+    const NftTransferWrapper = await deployer.deploy(
+        NftTransfer,
+        {},
+        config.contracts.ModuleRegistry,
+        config.modules.GuardianStorage
+    );
+    newModuleWrappers.push(NftTransferWrapper);  
```

Comment:
I feel that there is a bit of a mix of generic code and code specific to the NFT update. I would probably try to put the generic code (at least L69-L127, and possibly also the code in L51-L69) in one function that is called with the the names of the modules to enable, the names of the modules to disable, and a dictionary with the module names as keys and the moduleWrappers as values. Something like that...

Reply:
I fully agree, but I want to make sure the entire upgrade process is working end-to-end before investing time into extracting the generic code. I was planning to do that for the next upgrade (pnormally defi).